### PR TITLE
feat: update UI, deps, go version

### PR DESCRIPTION
**Pull request by caarlos0**
this cherry-picks/etc some commits from the [beta branch](https://github.com/charmbracelet/glow/compare/beta) into master.

Probably at least the spinner is broken 👀

### CODE REVIEW

File: ui/pager.go [hunk 1]
```diff
@@ -267,13 +312,13 @@ func (m pagerModel) update(msg tea.Msg) (pagerModel, tea.Cmd) {
 		}
 
 	case spinner.TickMsg:
-		if m.state == pagerStateStashing || m.spinner.Visible() {
+		if m.state == pagerStateStashing { // TODO: || m.spinner.Visible() {
```
Comment (caarlos0) [inline ui/pager.go:315]: @meowgorithm here I'll need some help probably, not sure how this worked before...

Comment (meowgorithm) [reply ui/pager.go:315]: Sounds good. If this PR is ready I can swoop in in a bit and take a look.

Comment (caarlos0) [reply ui/pager.go:315]: fixing some linting issues, but it should be good to go :)

**Comment by muesli** [comment]
Reviewing 🙌

**Pull request merged**